aleph/views/entities_api.py

```diff
@@ -576,3 +577,83 @@ def expand(entity_id):
             "results": results,
         }
     )
+
+
+@blueprint.route("/api/2/entities/<entity_id>/entitysets", methods=["GET"])
+def entity_entitysets(entity_id):
+    """Returns a list of entitysets which the entity has references in
+    ---
+    get:
+      summary: Shows EntitySets which reference the given entity
+      description: >-
+        Search for all entitysets which reference the given entity. The entity
+        sets can be filtered based on it's collection id, label, type or the
+        judgement of the entity within the set.
+      parameters:
+      - in: path
+        name: entity_id
+        required: true
+        schema:
+          type: string
+      - in: query
+        name: 'filter:type'
+        description: Restrict to a EntitySets of a particular type
+        required: false
+        schema:
+          items:
+            type: string
+          type: array
+      - in: query
+        name: 'filter:label'
+        description: Restrict to a EntitySets with a particular label
+        required: false
+        schema:
+          items:
+            type: string
+          type: array
+      - in: query
+        name: 'filter:judgement'
+        description: Restrict to a specific profile judgement
+        required: false
+        schema:
+          items:
+            type: string
+          type: array
+      - in: query
+        name: 'filter:collection_id'
+        description: Restrict to entity sets within particular collections
+        schema:
+          items:
+            type: string
+          type: array
+      responses:
+        '200':
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/EntitySet'
+          description: OK
+      tags:
+      - Entity
+      - Profiles
+    """
+    entity = get_index_entity(entity_id, request.authz.READ)
+
+    parser = QueryParser(request.args, request.authz)
+    collection_ids = [
+        cid
+        for cid in parser.filters.get("collection_id", [])
+        if request.authz.can(cid, request.authz.READ)
+    ]
+    judgements = parser.filters.get("judgement")
+    labels = parser.filters.get("label")
+    types = parser.filters.get("type")
+
+    entitysets = get_entitysets_by_entity(
```

Comment:
Maybe we can inline the query building to here? It's only 2, 3 lines and we can avoid one round of indirection.

Reply:
Sure thing!